app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingReservationPresenter.kt

```diff
@@ -0,0 +1,51 @@
+package com.example.parkingsystem.mvp.presenter
+
+import android.app.DatePickerDialog
+import android.app.TimePickerDialog
+import com.example.parkingsystem.mvp.contract.ParkingReservationContract
+import com.example.parkingsystem.util.Constants
+
+class ParkingReservationPresenter(
+    private val model: ParkingReservationContract.ParkingReservationModel,
+    private val view: ParkingReservationContract.ParkingReservationView
+) : ParkingReservationContract.ParkingReservationPresenter {
+
+    override fun onButtonParkingReservationPickerPressed(listener: DatePickerDialog.OnDateSetListener) {
+        model.setDateBeginPressed(view.getButtonPickerBegin())
+        view.showDatePickerDialog(listener)
+    }
+
+    override fun onDateSetPressed(year: Int, month: Int, dayOfMonth: Int, timeListener: TimePickerDialog.OnTimeSetListener) {
+        val sDate = "$dayOfMonth${Constants.SLASH}${month + Constants.ONE}${Constants.SLASH}$year"
+        val date = model.convertToCalendar(sDate, Constants.FORMAT_DATE)
+        if (model.getDateBeginPressed()) {
+            model.setDateBegin(date)
+        } else {
+            model.setDateEnd(date)
+        }
+        view.showTimePickerDialog(timeListener)
+    }
+
+    override fun onTimeSetPressed(hourOfDay: Int, minute: Int) {
+        val sTime = "$hourOfDay${Constants.TWO_POINTS}$minute"
+        val time = model.convertToCalendar(sTime, Constants.FORMAT_TIME)
+        if (model.getDateBeginPressed()) {
+            model.setTimeBegin(time)
+            view.enableDateEnd()
+            view.showDateAndTimeBegin(
+                model.getFormattedString(model.getDateBegin(), Constants.FORMAT_DATE),
```

Comment:
I think that the best approach is to have a method that unifies date and time begin and date and time end so you will ask the model only one calendar